### PR TITLE
feat: make `--gas-report` JSON output compatible

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Closes: https://github.com/foundry-rs/foundry/issues/8789

Part of meta-ticket https://github.com/foundry-rs/foundry/issues/8794

## Solution

Updates existing relevant tests to use Snapbox and adds additional tests for JSON output

Adds an additonal check on `--gas-report` when the `--junit` flag is specified as JUnit output is not supported

Adds a `silent` variable to mute unexpected output, this can be expanded in the future for other outputs that are mutually exclusive with the standard output

Each test suite ran with `--json` is written to an individual line to allow users to easily consume the output by parsing by line.

JSON formatted output looks like this:

```
{
   "gas":106715,
   "size":277,
   "functions":{
      "increment":{
         "increment()":{
            "calls":1,
            "min":43404,
            "mean":43404,
            "median":43404,
            "max":43404
         }
      },
      "number":{
         "number()":{
            "calls":257,
            "min":283,
            "mean":283,
            "median":283,
            "max":283
         }
      },
      "setNumber":{
         "setNumber(uint256)":{
            "calls":258,
            "min":23582,
            "mean":43384,
            "median":43542,
            "max":43866
         }
      }
   }
}
```

Compare to Markdown output:

```
| src/Counter.sol:Counter contract |                 |       |        |       |         |
|----------------------------------|-----------------|-------|--------|-------|---------|
| Deployment Cost                  | Deployment Size |       |        |       |         |
| 106715                           | 277             |       |        |       |         |
| Function Name                    | min             | avg   | median | max   | # calls |
| increment                        | 43404           | 43404 | 43404  | 43404 | 1       |
| number                           | 283             | 283   | 283    | 283   | 257     |
| setNumber                        | 23582           | 43224 | 43554  | 43866 | 258     |
```

### CODE REVIEW

**Review by grandizzy — APPROVED** [review]
lgtm!

File: crates/forge/tests/cli/cmd.rs [hunk 1]
```diff
@@ -1532,34 +1530,127 @@ contract ContractThreeTest is DSTest {
...
-        .get_output()
-        .stdout_lossy();
-    assert!(first_out.contains("foo") && first_out.contains("bar") && first_out.contains("baz"));
+        .stdout_eq(str![[r#"
+{"gas":103375,"size":255,"functions":{"foo":{"foo()":{"calls":1,"min":45387,"mean":45387,"median":45387,"max":45387}}}}
+{"gas":103591,"size":256,"functions":{"baz":{"baz()":{"calls":1,"min":260712,"mean":260712,"median":260712,"max":260712}}}}
+{"gas":103375,"size":255,"functions":{"bar":{"bar()":{"calls":1,"min":64984,"mean":64984,"median":64984,"max":64984}}}}
+
+
+"#]]);
```
Comment (DaniPopes) [inline crates/forge/tests/cli/cmd.rs:1571]: use .json_lines() on str

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/forge/tests/cli/cmd.rs (hunk 1 above)
Comment (zerosnacks) [reply crates/forge/tests/cli/cmd.rs]: Nice! Using `is_jsonlines()` (`.json_lines()` has been deprecated in favor of `is_jsonlines()`)

**Comment by zerosnacks** [comment]
ping @DaniPopes for final review

**Review by DaniPopes — APPROVED** [review]

**Comment by zerosnacks** [comment]
Hi @sakulstra thanks for your feedback, I've opened a new ticket here with your points as feature requests: https://github.com/foundry-rs/foundry/issues/9111 assigned to me

**Pull request merged**